shared/api/container.go

```diff
@@ -55,6 +55,11 @@ type ContainerPut struct {
 
 	// API extension: entity_description
 	Description string `json:"description" yaml:"description"`
+
+	// API extension: snapshot_expiry
+	// This field is used for `lxc config show` only, and is only shown for snapshots
+	// since regular containers cannot expire.
+	ExpiresAt *time.Time `json:"expires_at,omitempty" yaml:"expires_at,omitempty"`
```

Comment:
I think I'd prefer we don't do this kind of trickery and instead change `lxc config show` to not abuse ContainerPut for snapshots.

Reply:
We could just allow editing the entire snapshot config. That would make things pretty simple.

Reply:
Yeah, showing everything and discarding everything but expiry would be fine, I think that's what we do in a few other places.